src/common/aggregate/aggregator.js

```diff
@@ -82,12 +84,36 @@ export class Aggregator extends SharedContext {
     var hasData = false
     for (var i = 0; i < types.length; i++) {
       type = types[i]
-      results[type] = toArray(this.aggregatedData[type])
+      results[type] = this.toArray(type, this.aggregatedData[type])
+
       if (results[type].length) hasData = true
       delete this.aggregatedData[type]
     }
     return hasData ? results : null
   }
+
+  toArray (type, obj) {
+    if (typeof obj !== 'object') return []
+
+    return mapOwn(obj, (key, value) => this.getValue(key, value, type))
+  }
+
+  getValue (key, value, type) {
+    // err type can be decorated with an interaction ID if present
+    if (type !== 'err') return value
+    const shouldHold = this.checkInteractions(value) // add an interaction ID if exists
+    if (shouldHold) return
+    return value
+  }
+
+  checkInteractions (value) {
+    const spaFeature = getFeatureState({ agentIdentifier: this.sharedContext.agentIdentifier, featureName: FEATURE_NAMES.basicSpa })
+    const { shouldHold, interactions } = spaFeature?.hasInteraction?.({ timestamp: value?.metrics?.time?.t }) || {}
```

Comment:
Where is this `shouldHold`? the .hasInteraction only ever returns an object with just a single interactions property.